common/generate_bundle_base

```diff
@@ -9,13 +9,19 @@ declare -a overlay_opts=()
 # Catch any changes/additions to master opts
 update_master_opts ${MOD_PASSTHROUGH_OPTS[@]}
 
+type=$(get_cloud_type)
 vip_start=${MASTER_OPTS[VIP_ADDR_START]}
+cidr=""
 if [[ -z $vip_start ]] && [[ -e ~/novarc ]]; then
-    # prodstack
-    cidr=$(source ~/novarc; openstack subnet show subnet_${OS_USERNAME}-psd -c cidr -f value 2>/dev/null)
+    if [[ $type = openstack ]]; then
+        # prodstack
+        cidr=$(source ~/novarc; openstack subnet show subnet_${OS_USERNAME}-psd -c cidr -f value 2>/dev/null)
+    fi
     if [[ -z $cidr ]]; then
-        # stsstack
-        cidr=$(source ~/novarc; openstack subnet show ${OS_USERNAME}_admin_subnet -c cidr -f value 2>/dev/null)
+        if [[ $type = openstack ]]; then
+            # stsstack
+            cidr=$(source ~/novarc; openstack subnet show ${OS_USERNAME}_admin_subnet -c cidr -f value 2>/dev/null)
+        fi
         if [[ -n $cidr ]]; then
             vip_start=$(echo $cidr| sed -r 's/([0-9]+\.[0-9]+).+/\1/g').150.0
         fi
```

Comment:
I think all the above code should be under the 'openstack' if since it is only used in that case, correct? Then you wouldn't need the cidr="" assignment either.

Reply:
hi brian, I have added the following code for non-openstack cloud to improve this scene. thanks for the review.

elif [[ -z $vip_start  ]]; then
    echo "ERROR: No VIP start address provided with --vip-addr-start for the $type cloud."
exit 1